src/data-workspace/data-value-mutations/mutation-functions.js

```diff
@@ -28,6 +28,16 @@ const UPLOAD_FILE_MUTATION = {
     data: (data) => data,
 }
 
+export const createSharedMutateFn = (engine, mutationObj) =>
+    function mutateFn(variables) {
+        const { mutationKey } = this
+        // gather params from mutationKey
+        const { params } = mutationKey[1]
```

Comment:
```suggestion
        // A trick: get params for the mutation (de, co, ds, cc, cp) from the
        // `mutationKey`, which is set in the mutation options object and can
        // be accessed on `this` in the mutation function.
        // This saves passing `dataValueParams` everywhere
        const { mutationKey } = this
        // gather params from mutationKey
        const { params } = mutationKey[1]
```

Just an explanation suggestion for posterity